internal/controller/kuadrant_status_updater.go

```diff
@@ -145,6 +244,117 @@ func mtlsLimitador(kObj *kuadrantv1beta1.Kuadrant, state *sync.Map) *bool {
 	return ptr.To(kObj.IsMTLSLimitadorEnabled() && len(effectiveRateLimitPoliciesMap) > 0)
 }
 
+func (r *KuadrantStatusUpdater) resilienceCondition(topology *machinery.Topology) ([]*metav1.Condition, []string) {
+	create := make([]*metav1.Condition, 0)
+	remove := make([]string, 0)
+
+	kObj := GetKuadrantFromTopology(topology)
+	isConfigured := kObj.Spec.Resilience.IsRateLimitingConfigured()
+	if !isConfigured {
+		remove = append(remove, ResilienceInfoRRConditionType, ResilienceWarningRRConditionType, ResilienceInfoPDBConditionType, ResilienceInfoTSCConditionType)
+		return nil, remove
+	}
+
+	lObj := GetLimitadorFromTopology(topology)
+	if lObj == nil {
+		remove = append(remove, ResilienceInfoRRConditionType, ResilienceWarningRRConditionType, ResilienceInfoPDBConditionType, ResilienceInfoTSCConditionType)
+		return nil, remove
+	}
+
+	if lObj.Spec.Replicas != nil && *lObj.Spec.Replicas < LimitadorReplicas {
```

Comment:
have you considered adding some functions here such as something like `userModifiedReplicas(dep *Deployment, expected int)(bool, string)` where the bool represents if it user changed and the string the message to use?